src/sql/azdata.proposed.d.ts

```diff
@@ -432,6 +432,53 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+
+	export namespace diagnostics {
+		/**
+		 * Represents a diagnostics provider of accounts.
+		 */
+		export interface ErrorDiagnosticsProviderMetadata {
+			/**
+			 * The name of the provider (ex. a connection provider) that the diagnostics provider will handle errors for.
+			 */
+			id: string;
```

Comment:
Provider IDs aren't globally unique so not sure that the comment here is something you can reasonably guarantee. It's perfectly legal for something to register multiple different providers with the same ID - which in that case how will you know which one to use? Or is the plan just to take any providers with this ID and then pass stuff on to the diagnostic provider regardless?

@cheenamalhotra How are you thinking this is going to work?

Reply:
Exactly that, any providers with this ID will pass stuff onto the diagnostic provider for now.

Reply:
That's how all our services do, accept provider ID registration and that can be done any number of times by anyone. So its something we need to address globally, not in scope of this PR.

Reply:
This is the only provider we have that interacts with OTHER providers though. All the other ones don't need to (and shouldn't) be globally unique. They just need to be unique amongst the providers registered with them. 

But because this is a generic "diagnostics" provider then the way that you've designed it seems to be intended to use with other types of providers, which has the issues I outlined above. What happens when you want to add error handling for the resource provider services as well? 

So yes, not an immediate concern because you only deal with one provider (connection) currently, but given all the work you've put into making this generic I assume you still intend on supporting other types of providers later on so something you should be thinking about how to address when you get there.

Reply:
Let's discuss it here: https://github.com/microsoft/azuredatastudio/issues/21785

Reply:
Regarding error diagnostics service:

> This is the only provider we have that interacts with OTHER providers though.

ErrorDiagnosticsService is a service that providers register to. e.g. MSSQL. We're not creating a provider for ErrorDiagnostics, but other providers will register with their metadata to be able to handle APIs from this service.

> What happens when you want to add error handling for the resource provider services as well?

When a provider registers for error diagnostics service, it will support error handling APIs as applicable.

I think terms are confusing here, Error Diagnostics is a Service interface that providers can implement and register with, e.g. ObjectExplorerService, AccountManagementService, and when we need to call an API from the service, we look for service registration from the resp. provider and delegate call accordingly.

That's all that this is. It's purpose is to be the go-to service that will delegate error handling from SQL workbench, so resp. providers can be notified. These are the same providers that support connection, OE, Azure login etc.

Reply:
> What happens when you want to add error handling for the resource provider services as well?

Resource Provider is a separate provider (id: 'Microsoft.Azure.SQL.ResourceProvider'), so it's handler will have to be registered with a different implementation, such that if SQL workbench wants to communicate with "Microsoft.Azure.SQL.ResourceProvider" provider when an error occurs, it's implementation will handle it respectively, and if custom APIs are needed, they would be added to ErrorDiagnosticService as needed.